tools/windows/install-help/cal/customaction.h

```diff
@@ -16,14 +17,51 @@ DWORD DeleteUser(const wchar_t* host, const wchar_t* name);
 
 bool AddPrivileges(PSID AccountSID, LSA_HANDLE PolicyHandle, LPCWSTR rightToAdd);
 bool RemovePrivileges(PSID AccountSID, LSA_HANDLE PolicyHandle, LPCWSTR rightToAdd);
-int EnableServiceForUser(CustomActionData& data, const std::wstring& service);
+int EnableServiceForUser(PSID sid, const std::wstring& service);
 DWORD AddUserToGroup(PSID userSid, wchar_t* groupSidString, wchar_t* defaultGroupName);
 DWORD DelUserFromGroup(PSID userSid, wchar_t* groupSidString, wchar_t* defaultGroupName);
 bool InitLsaString(
 	PLSA_UNICODE_STRING pLsaString,
 	LPCWSTR pwszString);
 
-PSID GetSidForUser(LPCWSTR host, LPCWSTR user);
+struct SidResult
+{
+    sid_ptr Sid;
+    std::wstring Domain;
+    DWORD Result;
+
+    SidResult(DWORD result)
+    : Result(result)
+    {}
+
+    SidResult(sid_ptr &sid, std::wstring const &domain, DWORD result)
+    : Sid(std::move(sid))
+    , Domain(domain)
+    , Result(result)
+    {
+        
+    }
+
+    SidResult(SidResult const &) = delete;
+
+    SidResult(SidResult && other) noexcept
```

Comment:
do I read this right that doing a copy invalidates the source?

so 
`SidResult a(/* some stuff */)`
SidResult b

b = a; // results in `a` no longer containing the sid?
`